docs/provisioners/overview.md

```diff
@@ -0,0 +1,142 @@
+# Provisioners
+
+The following subdirectories contain concrete Provisioner implementations.
+
+- [plain](plain.md) - provisions `plain+v0` k8s bundles
+- [registry](registry.md) - provisions `registry+v1` OLM bundles
+- [helm](helm.md) - provisions `helm+v3` helm bundles-
```

Comment:
This should fix https://github.com/operator-framework/rukpak/issues/496 too.

Reply:
Linked that issue to be closed by this PR.